client/src/components/InputTextField/InputTextField.js

```diff
@@ -0,0 +1,143 @@
+      return (
+        <div className={this.state.classList}>
+          <input type="text" name="name" placeholder="City of Residence" />
+        </div>
+      );
+    }
+    if (this.props.type === "Age") {
+      return (
+        <div className={this.state.classList}>
+          <input type="text" name="name" placeholder="What is your age?" />
+        </div>
+      );
+    }
+    if (this.props.type === "Dream") {
+      return (
+        <div className={this.state.classList}>
+          <input type="text" name="name" placeholder="What is your dream destination(s)?" />
+        </div>
+      );
+    }
+    if (this.props.type === "First Name") {
+      return (
+        <div className={this.state.classListSmall}>
+          <input type="text" name="name" placeholder="First Name" />
+        </div>
+      );
+    }
+    if (this.props.type === "Last Name") {
+      return (
+        <div className={this.state.classListSmall}>
+          <input type="text" name="name" placeholder="Last Name" />
+        </div>
+      );
+    }
+    if (this.props.type === "Email Small") {
+      return (
+        <div className={this.state.classListSmall}>
+          <input type="text" name="email" placeholder="Email" onChange={this.changeHandler} />
+        </div>
+      );
+    }
+    if (this.props.type === "Password") {
+      return (
+        <div className={this.state.classList}>
+          <input type="password" name="name" placeholder="Password" />
+        </div>
+      );
+    }
+    if (this.props.type === "Confirm Password") {
+      return (
+        <div className={this.state.classList}>
+          <input type="password" name="name" placeholder="Confirm Password" />
+        </div>
+      );
+    }
+    if (this.props.type === "Search Directory") {
+      return (
+        <div className={this.state.classListSearch}>
+          <svg viewBox="-4 -2 24 24">
+            <path
+              fill="#bababa"
+              d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
+            />
+          </svg>
+          <input type="text" name="name" placeholder="Search My Directory" />
+        </div>
+      );
+    }
+    if (this.props.type === "Search Trips") {
+      return (
+        <div className={this.state.classListSearch}>
+          <svg viewBox="-4 -2 24 24">
+            <path
+              fill="#bababa"
+              d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
+            />
+          </svg>
+          <input type="text" name="name" placeholder="Search Trips" />
+        </div>
+      );
+    }
+  }
+}
```

Comment:
Can you simplify this? I don't think we need any of these conditionals. `type`, `name`, and `placeholder`, should all be passed down as props. As should `onChange`. You'll also need a prop for the icon (see #120) and the size

Reply:
Makes sense, refactoring everything now